### PR TITLE
Add onnx config for RoFormer

### PR DESCRIPTION
**Pull request by skrsna**
# What does this PR do?

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case. 
   - #16308 
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- ~[ ] Did you write any new necessary tests?~

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by ChainYo** [comment]
Hey nice PR! Did you try to convert one RoFormer model with your add ?
If so you could add the converted model to the [ONNXConfig for all](https://huggingface.co/OWG) organization, it would be awesome!

File: src/transformers/onnx/features.py [hunk 1]
```diff
@@ -313,6 +314,13 @@ class FeaturesManager:
             "question-answering",
             onnx_config_cls=Data2VecTextOnnxConfig,
         ),
+        "roformer": supported_features_mapping(
+            "default",
+            "masked-lm",
+            "sequence-classification",
+            "token-classification",
```
Comment (ChainYo) [inline src/transformers/onnx/features.py:321]: It seems that `RoFormer` also includes `causal-lm`, `multiple-choice` and `question-answering`.

File: src/transformers/models/roformer/configuration_roformer.py [hunk 2]
```diff
@@ -131,3 +135,16 @@ def __init__(
         self.layer_norm_eps = layer_norm_eps
         self.rotary_value = rotary_value
         self.use_cache = use_cache
+
+
+class RoFormerOnnxConfig(OnnxConfig):
+    @property
+    def inputs(self) -> Mapping[str, Mapping[int, str]]:
+        dynamic_axis = {0: "batch", 1: "sequence"}
```
Comment (ChainYo) [inline src/transformers/models/roformer/configuration_roformer.py:143]: You also need `multiple-choice` dynamic_axis like this: https://github.com/huggingface/transformers/pull/16859/files#diff-b111e7d3999d1220129c285224b7c0100290b6c17534dad32cd4e0b869b17fc0R150

Comment (skrsna) [reply src/transformers/models/roformer/configuration_roformer.py:143]: Thanks, I didn't look into detail about the supported roformer features, I'll add that now.

**Comment by skrsna** [comment]
> Hey nice PR! Did you try to convert one `RoFormer` model with your add ? If so you could add the converted model to the [ONNXConfig for all](https://huggingface.co/OWG) organization, it would be awesome!

```
python -m transformers.onnx -m junnyu/roformer_chinese_sim_char_ft_small onnx/model.onnx
Validating ONNX model...
	-[✓] ONNX model output names match reference model ({'last_hidden_state'})
	- Validating ONNX Model output "last_hidden_state":
		-[✓] (2, 8, 384) matches (2, 8, 384)
		-[✓] all values close (atol: 1e-05)
All good, model saved at: onnx/model.onnx
``` 
:)

**Comment by ChainYo** [comment]
It seems that there is 2x `sequence-classification` for features

> All good, model saved at: onnx/model.onnx

Nice news!

File: src/transformers/onnx/features.py [hunk 3]
```diff
@@ -313,6 +314,18 @@ class FeaturesManager:
...
         ),
+        "roformer": supported_features_mapping(
+            "default",
+            "masked-lm",
+            "causal-lm",
+            "sequence-classification",
+            "token-classification",
+            "multiple-choice",
+            "question-answering",
+            "sequence-classification",
```
Comment (ChainYo) [inline src/transformers/onnx/features.py:325]:
```suggestion
```

Comment (ChainYo) [reply src/transformers/onnx/features.py:325]: Could you remove the duplicate please @skrsna :hugs:

**Review by lewtun — COMMENTED** [review]
Thanks for adding RoFormer @skrsna and thank you to @ChainYo for the review! 

I agree with @ChainYo's suggestions and would also like to see this included in the slow tests if possible. Could you add a roformer checkpoint to `text_onnx_v2.py` please, e.g. maybe this works well: https://huggingface.co/junnyu/roformer_chinese_base

Then please run 

```
RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -s -k "roformer"
```

**Comment by skrsna** [comment]
> Thanks for adding RoFormer @skrsna and thank you to @ChainYo for the review!
> 
> I agree with @ChainYo's suggestions and would also like to see this included in the slow tests if possible. Could you add a roformer checkpoint to `text_onnx_v2.py` please, e.g. maybe this works well: https://huggingface.co/junnyu/roformer_chinese_base
> 
> Then please run
> 
> ```
> RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -s -k "roformer"
> ```

Thanks for the review @ChainYo and @lewtun, I added roformer to the onnx test case 

```
tests/onnx/test_onnx_v2.py::OnnxExportTestCaseV2::test_pytorch_export_65_roformer_causal_lm If you want to use `RoFormerForCausalLM` as a standalone, add `is_decoder=True.`
PASSED
tests/onnx/test_onnx_v2.py::OnnxExportTestCaseV2::test_pytorch_export_66_roformer_default PASSED
tests/onnx/test_onnx_v2.py::OnnxExportTestCaseV2::test_pytorch_export_67_roformer_masked_lm PASSED
tests/onnx/test_onnx_v2.py::OnnxExportTestCaseV2::test_pytorch_export_68_roformer_multiple_choice PASSED
tests/onnx/test_onnx_v2.py::OnnxExportTestCaseV2::test_pytorch_export_69_roformer_question_answering PASSED
tests/onnx/test_onnx_v2.py::OnnxExportTestCaseV2::test_pytorch_export_70_roformer_sequence_classification PASSED
tests/onnx/test_onnx_v2.py::OnnxExportTestCaseV2::test_pytorch_export_71_roformer_token_classification PASSED

```

**Review by lewtun — APPROVED** [review]
Great, thanks for adding the slow test and checking that it works!

This LGTM so gently pinging @sgugger for final approval

**Review by sgugger — APPROVED** [review]
Thanks for your PR!

**Pull request merged**